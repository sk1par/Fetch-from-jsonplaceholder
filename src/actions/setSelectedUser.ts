import IUser from "../models/user.interface";

const setSelectedUser= (user?: IUser) => {
    return {
        type: 'SELECTED_USER',
        payload: user
    }
}

export default setSelectedUser;