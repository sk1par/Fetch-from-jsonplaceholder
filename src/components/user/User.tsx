import { useDispatch } from 'react-redux';
import { setSelectedUser } from '../../actions';
import IUser from '../../models/user.interface';

interface UserInterface {
    user: IUser;
}

const User = ({ user }: UserInterface) => {
    const dispatch = useDispatch();

    const selectUser = () => {
        dispatch(setSelectedUser(user));
    };

    return (
        <div className="card" onClick={selectUser} data-auto-id='user-div'>
            <div className="card-body">
                <h5 className="card-title">{user?.name}</h5>
                <h6 className="card-subtitle mb-2 text-muted">
                    {user?.email}
                </h6>
            </div>
        </div>
    )
}

export default User
