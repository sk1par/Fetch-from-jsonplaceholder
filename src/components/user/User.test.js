import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import User from './User';
import { useSelector } from 'react-redux';
import  setSelectedUser from '../../actions/setSelectedUser';


Enzyme.configure({ adapter: new Adapter() });
jest.mock('react-redux', () => ({
    useDispatch: () => (fn) => () => fn(),
    useSelector: jest.fn(() => []),
}));
jest.mock('../../actions/setSelectedUser', () => {
    const setSelectedUser = jest.fn();
    return setSelectedUser;
});

const getInitialObject = () => [{
    email: 'test@abv.bg',
    id: 1,
    name: 'test',
    phone: '123132',
    username: 'skipar',
    website: 'abv.bg'
}];

it('should render div with user info correctly', () => {
    const value = getInitialObject();
    useSelector.mockReturnValueOnce([value]);
    const wrapper = shallow(<User user={value} />);
    const item = wrapper.find('[data-auto-id="user-div"]');
    item.simulate('click');
    expect(setSelectedUser).toHaveBeenCalledTimes(1);
    expect(setSelectedUser).toHaveBeenCalledWith(value);
    expect(wrapper).toMatchSnapshot();
});
