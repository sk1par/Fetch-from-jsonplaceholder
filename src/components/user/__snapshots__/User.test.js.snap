// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render div with user info correctly 1`] = `ShallowWrapper {}`;
